nix/default.nix

```diff
@@ -119,7 +119,7 @@ let
 
       substituteInPlace \
           ulauncher/modes/extensions/extension_controller.py \
-        --replace-fail '"PYTHONPATH": paths.APPLICATION,' '"PYTHONPATH": ":".join(sys.path),'
+        --replace-fail 'paths.APPLICATION,' '":".join(sys.path),'
```

Comment:
I haven't used nix before.
As I understand it helps setting up a local dev environment? Curious, what was the main challenge that nix solves for you?

Reply:
It's a package manager: https://www.youtube.com/watch?v=FJVFXsNzYZQ. I don't use it. It was contributed by @nazarewk.

Reply:
I definitely remember this PYTHONPATH patch was something we talked about when merging it originally. I definitely would want confirmation from some nix user about this.

Reply:
I don't remember the exact details, but `paths.APPLICATIONS` assumes Ulauncher lives inside a single directory (probably `~/.local/share`).

In Nix it's not a single directory, but a combination of **many** `/nix/store/` PYTHONPATHs: one per dependency. As far as I remember, there was nothing there except for `ulauncher` module without this patch.

Reply:
Again I don't remember the details apart from the non-Nix way having a good reason to be there, but this seems to be the whole discussion from when I wanted to "upstream" the change instead of using a patch: https://github.com/Ulauncher/Ulauncher/pull/1306